packages/pass-style/src/types.d.ts

```diff
@@ -80,20 +80,54 @@ export type PassByRef =
  * using 'slots').
  */
 export type Passable<
-  PC extends PassableCap = PassableCap,
+  R extends RemotableObject = RemotableObject,
   E extends Error = Error,
-> = void | Primitive | Container<PC, E> | PC | E;
-
-export type Container<PC extends PassableCap, E extends Error> =
-  | CopyArrayI<PC, E>
-  | CopyRecordI<PC, E>
-  | CopyTaggedI<PC, E>;
-interface CopyArrayI<PC extends PassableCap, E extends Error>
-  extends CopyArray<Passable<PC, E>> {}
-interface CopyRecordI<PC extends PassableCap, E extends Error>
-  extends CopyRecord<Passable<PC, E>> {}
-interface CopyTaggedI<PC extends PassableCap, E extends Error>
-  extends CopyTagged<string, Passable<PC, E>> {}
+  AllowPromise extends boolean = any,
```

Comment:
What does it mean to both type-constrain the type variable `AllowPromise` to `boolean` but give it a default of `any`?

Reply:
Curious why you didn't adopt the convention of the previous two type variables and write

```suggestion
  P extends Promise = Promise,
```
?

Reply:
Or, given the other changes

```suggestion
  P extends PromiseLike = PromiseLike,
```

Reply:
It's a bit of a hack in the type system. I need to conditionally include a `PromiseLike` in the union, or a `never`. But the `PromiseLike` needs to be parametrized with a specific constraint. As such this type parameter cannot be provided as a type (or inferred), but instead is just a flag to say "please allow promises in the union".

The `any` default is also a hack to make TS bail out when trying to type match. Without it we wouldn't be able to assign a `Passable` without promises (`AllowPromise=false`) to ones with it (or really places where we don't care). I don't fully understand why, this is where I ended at experimentally.